src/codec/RowWriterV2.cpp

```diff
@@ -16,6 +16,61 @@ namespace nebula {
 
 using nebula::cpp2::PropertyType;
 
+// Template functions for handling different types (e.g. strings, ints, floats) in lists and
+// collections
+template <typename Container>
+WriteResult writeContainer(const Container& container,
+                           Value::Type valueType,
+                           bool isSetType,
+                           std::unordered_set<std::string>& serializedStrings,
```

Comment:
Thank you for your suggestion. I have further improved it, but our implementation can only distinguish at runtime and cannot be done during static compilation.

Reply:
> Thank you for your suggestion. I have further improved it, but our implementation can only distinguish at runtime and cannot be done during static compilation.

We can determine its type outside the loop, instead of checking its type each time.
```
template <typename Type>
WriteResult write(...) {
    std::unordered_set<Type> serialized;
    if constexpr (std::is_same_v<Type, int>) {
        ...
    } else if constexpr (std::is_same_v<Type, std::string>) {
        ...
    } else if constexpr (std::is_same_v<Type, float>) {
        ...
    }
}

    switch (propType) {
        case LIST_STRING: {
            write<std::string>();
        }
        case LIST_INT: {
            write<int64_t>();
        }
    }
```